WalletWasabi.Fluent/Behaviors/ShowAttachedFlyoutWhenFocusedBehavior.cs

```diff
@@ -0,0 +1,133 @@
+		if (flyoutBase is null)
+		{
+			return;
+		}
+
+		var controller = new FlyoutShowController(AssociatedObject, flyoutBase).DisposeWith(disposable);
+
+		FocusBasedFlyoutOpener(AssociatedObject, flyoutBase, controller).DisposeWith(disposable);
+		IsOpenPropertySynchronizer(controller).DisposeWith(disposable);
+
+		// EDGE CASES
+		// Edge case when the Visual Root becomes active and the Associated object is focused.
+		ActivateOpener(AssociatedObject, visualRoot, controller).DisposeWith(disposable);
+		DeactivateCloser(visualRoot, controller).DisposeWith(disposable);
+
+		// This is a workaround for the case when the user switches theme. The same behavior is detached and re-attached on theme changes.
+		// If you don't close it, the Flyout will show in an incorrect position. Maybe bug in Avalonia?
+		if (IsFlyoutOpen)
+		{
+			controller.SetIsOpen(false);
+		}
+	}
+
+	private static IDisposable DeactivateCloser(Control visualRoot, FlyoutShowController controller)
+	{
+		return Observable.FromEventPattern(visualRoot, nameof(Window.Deactivated))
+			.Do(_ => controller.SetIsOpen(false))
+			.Subscribe();
+	}
+
+	private static IDisposable ActivateOpener(
+		IInputElement associatedObject,
+		Control visualRoot,
+		FlyoutShowController controller)
+	{
+		return Observable.FromEventPattern(visualRoot, nameof(Window.Activated))
+			.Do(
+				_ =>
+				{
+					if (associatedObject.IsFocused)
+					{
+						controller.SetIsOpen(true);
+					}
+				})
+			.Subscribe();
+	}
+
+	private IDisposable IsOpenPropertySynchronizer(FlyoutShowController controller)
+	{
+		return this.WhenAnyValue(x => x.IsFlyoutOpen)
+			.Do(controller.SetIsOpen)
+			.Subscribe();
+	}
+
+	private IDisposable FocusBasedFlyoutOpener(
+		IInteractive associatedObject,
+		FlyoutBase flyoutBase,
+		FlyoutShowController controller)
+	{
+		var currentPopupHost = Observable
+			.FromEventPattern(flyoutBase, nameof(flyoutBase.Opened))
+			.Select(_ => ((IPopupHostProvider) flyoutBase).PopupHost?.Presenter)
+			.WhereNotNull();
+
+		var associatedGotFocus = associatedObject.OnEvent(InputElement.GotFocusEvent).ToSignal();
+		var associatedLostFocus = associatedObject.OnEvent(InputElement.LostFocusEvent).ToSignal();
+		var popupGotFocus = currentPopupHost.Select(x => x.OnEvent(InputElement.GotFocusEvent)).Switch().ToSignal();
+		var popupLostFocus = currentPopupHost.Select(x => x.OnEvent(InputElement.LostFocusEvent)).Switch().ToSignal();
+
+		var isFocused = associatedGotFocus
+			.Merge(popupGotFocus).Select(_ => true)
+			.Merge(associatedLostFocus.Merge(popupLostFocus).Select(_ => false));
+
+		return isFocused
+			.Buffer(TimeSpan.FromSeconds(0.1))
+			.Where(focusedList => focusedList.Any())
+			.Select(focusedList => focusedList.Any(focused => focused))
+			.DistinctUntilChanged()
+			.ObserveOn(RxApp.MainThreadScheduler)
+			.Do(
+				isOpen =>
+				{
+					controller.SetIsOpen(isOpen);
+					IsFlyoutOpen = isOpen;
+				})
```

Comment:
Shouldn't this be enough? `IsOpenPropertySynchronizer` ensures that it will be set in the controller too.

```suggestion
			.Do(isOpen => IsFlyoutOpen = isOpen)
```

Reply:
You're so right. Thanks!